### PR TITLE
Feature/task 98 set target for reference only quick fix

### PR DESCRIPTION
**Pull request by Anna-Lena-DLR**
A targetForReferences is always generated but only when using the QuickFix "Generate Target for References" sets the targetForReferences as active target and then the acutal target that couldnt be resolved.

grafik

### CODE REVIEW

File: de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend [hunk 1]
```diff
@@ -177,11 +173,54 @@ class ReferencedTargetHelperTest {
...
+		}
+		folder2.create(IResource.NONE, true, null);
+		val bytes2 = "
+			Target targetForReferences {
+				
+			}".getBytes();
+		val sourceTarget2 = new ByteArrayInputStream(bytes2);
+		tmodelFile2.create(sourceTarget2, IResource.NONE, null);
+
+		Assert.assertNull(refTargetHelper.findTargetForReferencesFile(tmodelFile2, outputDirectory))
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend:224]: Why do we expect NULL here? Maybe add a message in the assert statement.... 
`Assert.assertNull("File does not exist so shoud be null", refTargetHelper.findTargetForReferencesFile(tmodelFile2, outputDirectory))`

File: de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend [hunk 2]
```diff
@@ -177,11 +173,54 @@ class ReferencedTargetHelperTest {
...
+
+		Assert.assertNull(refTargetHelper.findTargetFileInProject(tmodelFile2, outputDirectory))
+	}
+	
+	@Test
+	def void testFindTargetForReferencesFile() {
+		val outputDirectory = "./target"
+		val workspace = ResourcesPlugin.getWorkspace();
+		val root = workspace.getRoot();
+		val project  = root.getProject("testProject3");
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend:185]: Maybe use final varibales here 
`val PROJECT_NAME = "testProject"`

Why these numbers? Maybe better name them according to their distinguishing feature. -> ProjectWithoutTmodel, ProjectWithTmoldel...

File: de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend [hunk 3]
```diff
@@ -48,16 +51,31 @@ class OvertargetQuickfixProvider extends DefaultQuickfixProvider {
 	
 	@Fix(Diagnostic.LINKING_DIAGNOSTIC)
 	def fixCannotResolveReference(Issue issue, IssueResolutionAcceptor acceptor) {
-		acceptor.accept(issue, 'Generate Target for References', '', '', 
+		acceptor.accept(issue, 'Generate targetForReferences to resolve references', 'Generates a targetForReferences and sets it as active target. \n After that the actual target, which could not be resolved, is set as active target.', '',
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend:54]:
```suggestion
		acceptor.accept(issue, 'Use temporary target to to resolve TModel references', 'Generates a temporary target for resolving TModel references and sets it as active target. \n After resolving the references of this target, it is set as active target again.', '', 
```

Comment (franzTobiasDLR) [reply de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend:54]: I think no one will understand what a "targetForReferences" is and what it does.

File: de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend [hunk 4]
```diff
@@ -48,16 +51,31 @@ class OvertargetQuickfixProvider extends DefaultQuickfixProvider {
...
 						val input = ite.editorInput
-						new GenerationHandler().runGeneration(input);
+						genHandler.runGeneration(input);
+						
+						//find targetForReferences.target in directory and set it as active target
+						val fileEditorInput = input as IFileEditorInput
+						val file = fileEditorInput.file
+						val outputDirectory = genHandler.getOutputConfigurations(input)
+						val targetForReferencesFile = refTargetHelper.findTargetForReferencesFile(file, outputDirectory)
+						targetPlatHelper.asActiveTarget = targetForReferencesFile;
```
Comment (franzTobiasDLR) [inline de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend:74]: Is it also resolving the target here?

Comment (Anna-Lena-DLR) [reply de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend:74]: not here, but in the TargetPlatformHelper

**Pull request merged**